### ISSUE TITLE
LMF gives random output despite random_state being set.

### ISSUE THREAD

**Issue by icefire112233** [benfred/implicit#402]
The random state of LMF doesn't work as expected. It gives different output everytime.
```
model =  LogisticMatrixFactorization(factors = 5, random_state = 1)

R = eye(5)
model.fit(R,False)

print(model.item_factors.dot(model.user_factors.transpose()))
```

Output:
![image](https://user-images.githubusercontent.com/50484732/94104176-fe9d5880-fe68-11ea-9080-8e4b01bbc6b1.png)

**Comment by ita9naiwa** [comment]
Thanks for reporting! Ill see soon

**Comment by benfred** [comment]
This is related to multithreading and each thread getting assigned different rows on each run - if you passed in 'num_threads=1' to the model this issue wouldn't happen. I have a fix here https://github.com/benfred/implicit/pull/408

**Comment by benfred** [comment]
Fixed by #408
